src/motor_control.c

```diff
@@ -87,7 +87,17 @@ void motor_control_apply(MotorControl *mc, float abs_erpm, RunState state, float
         VESC_IF->mc_set_current(mc->requested_current);
     } else {
         // Brake logic
-        if (abs_erpm > ERPM_MOVING_THRESHOLD) {
+        if (mc->parking_brake_mode == PARKING_BRAKE_IDLE) {
+            // only evaluate parking brake each cycle if pbmode is IDLE
+            if ((state != STATE_RUNNING) &&
+                ((md->abs_erpm_smooth < 50) || (md->last_erpm_sign != md->erpm_sign))) {
```

Comment:
Please don't use brackets around expressions when they are "obviously" not necessary, they add visual clutter. What's obvious is obviously subjective, but these are very common cases and everyone should know the general precedence. If in doubt, just consult https://en.cppreference.com/w/c/language/operator_precedence.

The condition here should be:
```c
if (state != STATE_RUNNING && (md->abs_erpm_smooth < 50 || md->last_erpm_sign != md->erpm_sign)) {
```

Reply:
@surfdado's approach is faster to understand at a glance, by encapsulating each comparison inside of its own _intentional_ set of brackets.

Having everything inline including one set of brackets while omitting brackets deeper than 2 suggests they were missed.

I'd completely go the other direction and prefer that each comparison & group be its own indentation.

White space costs nothing, and only makes code easier for human eyes to interpret. <3

Reply:
Sorry, this is subjective, to me it makes it harder to interpret (because I need to carefully verify each bracket encapsulation is correct, sometimes even resort to counting them, whereas if they're not there, it's much clearer to me).

> White space costs nothing ...

Not true, space costs exactly that, space on the screen. Extra brackets make lines longer, may make some lines need to be wrapped, greatly reducing readability. Any redundant extra characters increase size while not increasing information, therefore reducing entropy.

(And this is as far as I want to go into this discussion :sweat_smile:)